src/main/java/io/github/bobocodebreskul/context/registry/AnnotatedBeanDefinitionReader.java

```diff
@@ -101,40 +104,66 @@ public final BeanDefinitionRegistry getBeanDefinitionRegistry() {
   }
 
-  private <T> void doRegisterBean(Class<T> beanClass, String name) {
-    log.debug("doRegisterBean method invoked: beanClass={}, name={}", beanClass.getName(), name);
+  private <T> void doRegisterBean(Class<T> beanClass, String beanName) {
+    log.debug("doRegisterBean method invoked: beanClass={}, name={}", beanClass.getName(), beanName);
     log.info("Registering the bean definition of class {}", beanClass.getName());
     // todo: create beanDefinitionValidator that validate things such as:
     //  bean name validity (not allowed characters), check for circular dependency
 
     var annotatedBeanDefinition = new AnnotatedGenericBeanDefinition(beanClass);
-    name = name != null ? name : generateBeanName(annotatedBeanDefinition, beanDefinitionRegistry);
-    annotatedBeanDefinition.setName(name);
+    beanName = beanName != null ? beanName : generateBeanName(annotatedBeanDefinition, beanDefinitionRegistry);
+    annotatedBeanDefinition.setName(beanName);
 
-    if (beanDefinitionRegistry.isBeanNameInUse(name)) {
+    if (beanDefinitionRegistry.isBeanNameInUse(beanName)) {
       log.error("The specified bean name is already in use");
       throw new DuplicateBeanDefinitionException(
-          "The bean definition with specified name %s already exists".formatted(name));
+          "The bean definition with specified name %s already exists".formatted(beanName));
     }
 
-    annotatedBeanDefinition.setScope(BeanDefinition.SINGLETON_SCOPE);
 
     if (ReflectionUtils.isAnnotationPresentForClass(Primary.class, beanClass)) {
-      log.trace("Found @Primary annotation on the beanName={}", name);
+      log.trace("Found @Primary annotation on the beanName={}", beanName);
       annotatedBeanDefinition.setPrimary(true);
     }
 
-    Constructor<?> beanConstructor = findBeanInitConstructor(beanClass, name);
+    setBeanDefinitionScope(beanClass, beanName, annotatedBeanDefinition);
+
+    Constructor<?> beanConstructor = findBeanInitConstructor(beanClass, beanName);
     log.debug("Constructor found for bean class [{}]: [{}]", beanClass.getName(), beanConstructor);
     annotatedBeanDefinition.setInitConstructor(beanConstructor);
     List<BeanDependency> dependencies = getConstructorBeanDependencies(beanConstructor);
     log.debug("{} dependencies found for beanClass={} with beanName={}",
-        dependencies.size(), beanClass.getName(), name);
+        dependencies.size(), beanClass.getName(), beanName);
     annotatedBeanDefinition.setDependencies(dependencies);
     annotatedBeanDefinition.setAutowireCandidate(
         isBeanAutowireCandidate(beanClass, beanDefinitionRegistry));
 
-    beanDefinitionRegistry.registerBeanDefinition(name, annotatedBeanDefinition);
+    beanDefinitionRegistry.registerBeanDefinition(beanName, annotatedBeanDefinition);
     log.trace("Registered bean definition: {}", annotatedBeanDefinition);
   }
+
+  private static <T> void setBeanDefinitionScope(Class<T> beanClass, String beanName,
```

Comment:
I recommend this method to return scope value and to set it in the method above, like other values.
First of all it will look and feel the same as other values.
Secondly by best practices it's better to return some strict value than providing action parameter (`annotatedBeanDefinition`) that could be changed in different ways in the method.

Reply:
done